tests/python/unittest/test_gluon_estimator.py

```diff
@@ -275,3 +276,45 @@ def test_context():
                                   loss=loss,
                                   metrics=metrics,
                                   context='cpu')
+
+
+def test_batch_size():
+    '''Test batch size'''
+    num_samples = 32
+
+    # No Data Loader
+    data = mx.nd.random.uniform(shape=(num_samples, 3, 28, 28))
+    label = mx.nd.random.randint(low=0, high=2, shape=(num_samples,))
+    data_iter = mx.io.NDArrayIter(data=data, label=label, batch_size=16)
+    net = get_model()
+    loss = mx.gluon.loss.L2Loss()
+    ctx = mx.cpu()
+    est = estimator.Estimator(net=net, loss=loss, context=ctx)
+    with assert_raises(ValueError):
+        est.fit(train_data=data_iter)
+
+    # Empty data loader
+    data = mx.nd.random.uniform(shape=(0,))
+    label = mx.nd.random.randint(low=0, high=2, shape=(0,))
+    batch_size = 2
+    data_arr = mx.gluon.data.dataset.ArrayDataset(data, label)
+    data_loader = mx.gluon.data.DataLoader(data_arr, batch_size=batch_size)
+    est = estimator.Estimator(net=net, loss=loss, context=ctx)
+    with assert_raises(ValueError):
+        est.fit(train_data=data_loader)
+
+    # Batch size less than context
+    ctx = [mx.gpu(0), mx.gpu(1), mx.gpu(2), mx.gpu(3)]
```

Comment:
This can be re-written as : ctx = [mx.gpu(i) for i in range(mx.context.num_gpus())]

Reply:
This specific testcase doesn't require the actual GPUs to execute, which means, the aim of this test case is to check that the batch size should be greater than the total number of context to fully utilize the system. The code doesn't execute the forward/backward pass since the assert statement is at the start of fit() method.

should this test be run only on multi-gpu context environment?

Reply:
I believe so, or you can do something like this : 
```ctx = [mx.cpu(0), mx.cpu(0), mx.cpu(0), mx.cpu(0)]```

That would make it GPU agnostic and would just check for your test case to validate whether ```batch_size > len(ctx)``` 

WDYT ?